packages/server/src/fhir/repo.ts

```diff
@@ -2234,7 +2243,10 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
   private async setCacheEntry(resource: Resource): Promise<void> {
     // No cache access allowed mid-transaction
     if (this.transactionDepth) {
-      await this.postCommit(() => this.setCacheEntry(resource));
+      const cachedResource = deepClone(resource);
```

Comment:
Ouch, I wish we didn't need to clone before caching...

Is this because of the potential for destructive operations during the processing of a `Bundle`? Or something similar?

Reply:
Yeah, the core issue here is that, in context of a PATCH, without the `X-Medplum: extended` header (i.e. `extendedMode` in the `Repository`), several fields are removed from `Resource.meta` at the end of `updateResourceImpl()` — before the post-commit hooks run.  At least the cloning should only ever happen in cases where we're inside of a transaction: for simple writes the existing object should not be cloned, and would be stored in the cache immediately